pyproject.toml

```diff
@@ -12,102 +7,144 @@ license = "MIT license"
 tldextract = "3.5.0"
-nanoid = "2.0.0"
+# rsa = "4.9  # awscli 1.22.38 depends on rsa<4.8
+typing-extensions = "4.12.2"
 unidecode = "1.3.8"
-more-itertools = "8.14.0"
-# PaaS
-awscli-cwlogs = "1.4.6"
-aws-embedded-metrics = "1.0.8"
 # Putting upgrade on hold due to new version introducing breaking changes
 Werkzeug = "3.0.4"
-MarkupSafe = "2.1.5"
-# REVIEW: v2 is using sha512 instead of sha1 by default (in v1)
-itsdangerous = "2.2.0"
-notifications-utils = { git = "https://github.com/cds-snc/notifier-utils.git", tag = "53.1.2" }
-
-# rsa = "4.9  # awscli 1.22.38 depends on rsa<4.8
-typing-extensions = "4.12.2"
-greenlet = "3.1.1"
-simple-salesforce = "^1.12.3"
-
 # Pinned dependencies
-certifi = "^2024.0.0" # pinned for security reasons: https://github.com/cds-snc/notification-api/security/dependabot/119
 idna = "2.10" # pinned to align with test moto dependency requirements (for <=2.9)
-flask-marshmallow = "0.14.0"
-aws-xray-sdk = "2.14.0"
 
 [tool.poetry.group.test.dependencies]
-flake8 = "6.1.0"
+coveralls = "3.3.1"
+freezegun = "1.5.1"
 gprof2dot = "2024.6.6"
-isort = "5.13.2"
+# used for creating manifest file locally
+jinja2-cli = { extras = ["yaml"], version = "0.8.2" }
+locust = "2.31.3"
 moto = "4.2.14"
-idna = "2.10"
+mypy = "1.5"
+networkx = "2.8.8" # not directly required, pinned by Snyk to avoid a vulnerability
 pytest = "7.4.4"
+pytest-cov = "3.0.0"
 pytest-env = "0.8.2"
 pytest-mock = "3.14.0"
-pytest-cov = "3.0.0"
-coveralls = "3.3.1"
+pytest-mock-resources = { extras = ["redis"], version = "2.12.0" }
 pytest-xdist = "2.5.0"
-freezegun = "1.5.1"
 requests-mock = "1.12.1"
+rfc3987 = "1.3.8"
+ruff = "^0.8.2"
 snakeviz = "2.2.0"
+sqlalchemy-stubs = "0.4"
+sqlalchemy2-stubs = "0.0.2a38"                                     # not directly required, pinned by Snyk to avoid a vulnerability
 # optional requirements for jsonschema
 strict-rfc3339 = "0.7"
-rfc3987 = "1.3.8"
-# used for creating manifest file locally
-jinja2-cli = { extras = ["yaml"], version = "0.8.2" }
-black = "23.12.1"
-locust = "2.31.3"
-mypy = "1.5"
-sqlalchemy-stubs = "0.4"
-sqlalchemy2-stubs = "0.0.2a38"
-networkx = "2.8.8" # not directly required, pinned by Snyk to avoid a vulnerability
-pytest-mock-resources = { extras = ["redis"], version = "2.12.0" }
 types-boto = "2.49.18.9"
 types-mock = "4.0.15.2"
 types-python-dateutil = "2.9.0.20240316"
 types-pytz = "2022.7.1.2"
-types-requests = "2.32.0.20240712"
 types-redis = "4.6.0.20240425"
+types-requests = "2.32.0.20240712"
+
+
+[tool.ruff]
+target-version = "py310"
+exclude = [
+    "venv*",
+    "__pycache__",
+    "node_modules",
+    "cache",
+    "migrations",
+    "build",
+]
+extend-include = ['(app|migrations|tests)/.*\.pyi?$']
+src = ["app", "migrations", "tests"]
+# Ruff formatter will wrap lines at a length of 130 characters.
+line-length = 130
+indent-width = 4
+
+[tool.ruff.lint]
+select = [
+    # PyFlakes
+    "F",
+    # Pycodestyle
+    "E",
+    "W",
+    # isort
+    "I001",
+    "I002",
```

Comment:
It would be nice to have what each of these are for on the right side.

Reply:
Documented!